tests/test_host.py

```diff
@@ -0,0 +1,68 @@
+import pytest
+
+from illuminatio.host import (
+    ClusterHost,
+    ConcreteClusterHost,
+    ExternalHost,
+    LocalHost,
+    GenericClusterHost,
+    Host,
+)
+
+
+@pytest.mark.parametrize(
+    "identifier,expected_host",
+    [
+        pytest.param("localhost", LocalHost(), id="LocalHost"),
+        pytest.param(
+            "127.0.0.1", ExternalHost("127.0.0.1"), id="ExternalHost with localhost IP"
+        ),
+        pytest.param(
+            "123.456.789.148",
+            ExternalHost("123.456.789.148"),
+            id="ExternalHost with long IP",
+        ),
+        pytest.param(
+            "1.2.3.4", ExternalHost("1.2.3.4"), id="ExternalHost with short IP"
+        ),
+        pytest.param(
+            "default:nginx-23429-asdf",
+            ConcreteClusterHost("default", "nginx-23429-asdf"),
+            id="Simple ConcreteClusterHost",
+        ),
+        pytest.param(
+            "default:test=test",
+            ClusterHost("default", {"test": "test"}),
+            id="Simple ClusterHost",
+        ),
+        pytest.param(
+            "illuminatio-inverted-default:illuminatio-inverted-test.io/test-123_XYZ=test_456-123.ABC",
+            ClusterHost(
+                "illuminatio-inverted-default",
+                {"illuminatio-inverted-test.io/test-123_XYZ": "test_456-123.ABC"},
+            ),
+            id="ClusterHost containing all allowed label characters",
+        ),
+        pytest.param(
+            "test=test:test=test",
+            GenericClusterHost({"test": "test"}, {"test": "test"}),
+            id="Simple GenericClusterHost",
+        ),
+        pytest.param(
+            "illuminatio-inverted-test.io/test-123_XYZ=test_456-123.ABC:illuminatio-inverted-test.io/test-123_XYZ=test_456-123.ABC",
```

Comment:
the linter complains about this line: https://travis-ci.org/github/inovex/illuminatio/jobs/718928535#L331